syserol/tensor.py

```diff
@@ -191,16 +234,23 @@ def perform_CMTF(tOrig=None, r=6):
     # Precalculate the missingness patterns
     uniqueInfo = [np.unique(np.isfinite(B.T), axis=1, return_inverse=True) for B in unfolded]
 
+    # get unique days into vector format for continuous solve
+    days = dayLabels()
+    # initialize parameter matrix
+
     for ii in range(2000):
         # PARAFAC on all modes
         for m in range(0, len(tFac.factors)):
             kr = khatri_rao(tFac.factors, skip_matrix=m)
             tFac.factors[m] = censored_lstsq(kr, unfolded[m].T, uniqueInfo[m])
         
+        # for final (continuous) dimension, solve for P and recalculate continuous factor
+        tFac.factors[m] = continue_solve(days, tFac.factors[m])
+
         if ii % 2 == 0:
             R2X_last = tFac.R2X
             tFac.R2X = calcR2X(tFac, tOrig)
-            assert tFac.R2X > 0.0
```

Comment:
Are you finding you get negative R2X values?

Reply:
Seems like for the polynomial solve, we don't get negative R2X values.